### PR TITLE
Compatibility with Arduino 1.6.0

### PR DESCRIPTION
**Pull request by patrickelectric**
Now the IDE can compile and use Arduino version 1.6.0
TODO: Need to test upload with real boards

### CODE REVIEW

**Comment by mupuf** [comment]
Thanks Patrick!

I will have a look at your patches soon. However, could you fix the commits instead of adding more commits later to clean the cruft introduced by previous commits? I mean, the last 5 patches are unacceptable because they make it hard for people to study the changes.

If you wonder how to do this with git, you may be interested in looking into git rebase -i, that will allow you to squash, edit or reword commits :)

**Comment by patrickelectric** [comment]
No problem, sorry for that !

**Comment by mupuf** [comment]
If you keep on sending patches like that and if I like where you are going (which I so far do!), you may very well get commit rights :) So, bear with me while I annoy you with what seems like little issues :)

**Comment by patrickelectric** [comment]
Ty for the comment @mupuf , I believe that everything is alright now =D

**Comment by mupuf** [comment]
Thanks Patrick! I will have a look at them tonight! I did not notice you were so quick to fix your patches!

File: IDEApplication.cpp [hunk 1]
```diff
@@ -138,8 +138,8 @@ void IDEApplication::initializePlugins()
 void IDEApplication::initializeTranslator()
 {
     if(! mTranslator.load("arduide_" + QLocale::system().name(), TRANSLATION_PATH))
-     {
-      qDebug() << "loading translator failed.";
-     }
+    {
+        qDebug() << "loading translator failed.";
+    }
```
Comment (mupuf) [inline IDEApplication.cpp:143]: This change is unrelated, this is something one should generally avoid.

No problem for this time though.

File: env/Board.cpp [hunk 2]
```diff
@@ -60,31 +60,32 @@ void Board::listBoards()
             while (! boardsFile.atEnd())
             {
                 QString line = QString::fromLocal8Bit(boardsFile.readLine()).trimmed();
-	        //QString line = QString::fromLocal8Bit(boardsFile.readLine().simplified());
+
```
Comment (mupuf) [inline env/Board.cpp]: There should not be any trailing spaces, you can configure your editor to do just that :)

If I merge this series, you will be encouraged to write another patch to remove them all :p

**Comment by mupuf** [comment]
I merged it and fixed a few things related to the build system.

I also successfully tested the new code on my arduino Leonardo. In the process, I found out that we also need to require avrdude, so I updated Arch's package.

Looking forward to seeing more changes coming from you!

File: env/Board.cpp (hunk 2 above)
Comment (patrickelectric) [inline env/Board.cpp]: HOoo, sorry for that, I forgot to configure trailing spaces :/.

**Comment by patrickelectric** [comment]
Ty @mupuf !
In this moment I am trying to use QT5 in place of QT4, maybe in this weekend its finished :pray:

**Comment by mupuf** [comment]
Yepee! I started many times to do this but always stopped midway. Now that Qt5 is pretty well supported everywhere, I guess grantlee also supports qt5 so it sounds like a good time!

**Comment by mupuf** [comment]
A bug introduced by this patch series:
- https://github.com/mupuf/arduide/issues/4

Oh, and I also realized that the documentation was not available anymore. If you have some time to fix these little issues, I would appreciate it :)

**Comment by patrickelectric** [comment]
NP, I'm busy right now but I'll solve it.
I am compiling the QT5 version in my machine, in the place of grantlee it's possible to use qml of qt.
In the next week I will try to upload with some boards like uno, mini and nano.
